### DISCUSSION TITLE
FAQ

### DISCUSSION THREAD

**Discussion by natintosh** [natintosh/intl-phone-number-input#159] [Q&A] [answered]
Hello there, thanks for using this package. Here I'll be answering some frequent questions and concerns. If you need any help, feel free to drop a comment.

**Comment by abeticlairon** [comment, marked as answer]
Hello, is there a way to set an initial country? It always shows Afghanistan as the initial one

**Reply by zeusbaba** [reply]
yes, there is a parameter you can use while initializing, see example 
```
import 'package:intl_phone_number_input/intl_phone_number_input.dart'  as phoneInput;

phoneInput.InternationalPhoneNumberInput(
initialValue: phoneInput.PhoneNumber(isoCode: 'NO'),  
// ... set other params
);
```

**Reply by sawirricardo** [reply]
Hi, I've encountered this, the country initial is correct, yet the phone number didn't show up 🤔

**Reply by sawirricardo** [reply]
I think there's a bit confusion. I tried to set the initialValue and hoping that the text also show up. But actually, the phone number field are shown with TextEditingController().

**Reply by KibromHS** [reply]
It won't change to another country after initializing it with one country isoCode

**Comment by haroldmeza** [comment]
Hi, i receive a error when i try to write a phone number, the error is throw by message_codecs.dart line 581, 

This is my pubspec.yaml

version: 1.0.0+1

environment:
  sdk: ">=2.12.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter


  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.2
  intl_phone_number_input: ^0.7.0+1

**Comment by keshav-drupalchamp** [comment]
I am using this in web flutter. 
My SDK settings are:-
 
environment:
  sdk: ">=2.8.0 <3.0.0"

dev_dependencies:
  intl_phone_number_input: ^0.7.0+0.1


Run:-` flutter pub get`
Getting this error in return

[xyz] flutter pub get
Error detected in pubspec.yaml:
Unexpected child "intl_phone_number_input" found under "flutter".
Please correct the pubspec.yaml file at C:\src\xyz\pubspec.yaml
exit code 1

I am getting What the problem is actually here?

While using this project in separate alone clone after Github, working fine. But as I am using it in my project that's the error is coming.

Please suggest me.

**Comment by FOFE-DEV** [comment]
hello please how  to get iso2  or country name of selected country?

**Reply by MIDEV-05** [reply]
Hi ! I'm also looking for how to achieve this.
Did you somehow manage to get this values ?

**Reply by blamath20** [reply]
Same here. Any idea how to set the iso code using default locale?

**Reply by BastianValencia** [reply]
alguien pudo obtener el nombre del pais seleccionado?

**Comment by aatikakhan** [comment]
hi, 
Is intl_phone_number_input package support local language or it is only in english? 

Want to translate this hint text on search bar in other language
![Uploading selectCountry.jpeg…]()

**Reply by natintosh** [reply]
Oh,

Right now the search bar only supports English as it's default, but...

You can set your default hint text by updating the searchBoxDecoration, which accepts `InputDecoration`.

**Reply by aatikakhan** [reply]
It's not working 
```
searchBoxDecoration: InputDecoration(
                        hintText: 'Search by country name or dial code'.tr(),
                      ),
```

**Comment by Vladislav0777** [comment]
Hello! How do I CustomDecoration?  as in the example

**Reply by mnnlthmpsn** [reply]
> Hello! How do I CustomDecoration? as in the example

`inputDecoration: InputDecoration(
                            focusedBorder: UnderlineInputBorder(
                                borderSide: BorderSide(
                                    color:
                                        gSecondaryTextColor.withOpacity(.5),
                                    width: .5)),
                            enabledBorder: UnderlineInputBorder(
                                borderSide: BorderSide(
                                    color:
                                        gSecondaryTextColor.withOpacity(.5),
                                    width: .5)),
                            errorBorder: UnderlineInputBorder(
                                borderSide:
                                    BorderSide(color: Colors.red, width: .5)),
                            focusedErrorBorder: UnderlineInputBorder(
                                borderSide: BorderSide(
                                    color: Colors.red, width: .5)))`

**Comment by arojas90** [comment]
Hello, is there a way to assign a custom order to the list of countries?

**Comment by Coder-Manuel** [comment]
Hello how do you set the dropdown button to appear as in the custom decoration.....

**Comment by smh821025** [comment]
Hello, guys, I want to change the layout of the area icon, Put the icon in front of the text box.how to do?
<img width="260" alt="图片" src="https://user-images.githubusercontent.com/13348615/146917147-7df5dbb8-481d-428e-b90d-6faa4b706560.png">
Thanks

**Reply by Mohamed-Reda1010** [reply]
Use Directionality , set textDirection: TextDirection.ltr,

Example :

 Directionality buildPhoneFormField() {
    return Directionality(
      textDirection: TextDirection.ltr,
      child: InternationalPhoneNumberInput(
        textFieldController: phoneController,
        spaceBetweenSelectorAndTextField: 0,
        selectorConfig: const SelectorConfig(
          selectorType: PhoneInputSelectorType.BOTTOM_SHEET,
          showFlags: true,
          useEmoji: true,
        ),
        ignoreBlank: false,
        autoValidateMode: AutovalidateMode.always,
        selectorTextStyle: const TextStyle(color: Colors.black),
        initialValue: number,
        formatInput: false,
        keyboardType:
            const TextInputType.numberWithOptions(signed: true, decimal: true),
        inputBorder: const OutlineInputBorder(),
        onInputChanged: (PhoneNumber value) {
          String phoneWithoutCode = phoneNumberWithoutCountryCode(
              phone: value.phoneNumber!, countryCode: value.dialCode!);
          if (phoneWithoutCode.isNotEmpty) {
            removeError(error: kPhoneNumberNullError);
            //removeError(error: kPhoneNumberNullError);
          }

          if (phoneWithoutCode.length >= minPhoneLength) {
            removeError(error: kPhoneNumberShortError);

            // removeError(error: kPhoneNumberShortError);
          }

          phone = value.phoneNumber.toString();
          phoneCountryCode = value.dialCode;
          phoneWithoutCountryCode = phoneWithoutCode;
        },
        validator: (value) {
          if (value!.isEmpty) {
            addError(error: kPhoneNumberNullError);
            return "";
          } else if (value.length < minPhoneLength) {
            addError(error: kPhoneNumberShortError);
            return "";
          }
          return null;
        },

        onSaved: (PhoneNumber newValue) {
          phone = newValue.phoneNumber.toString();
          phoneCountryCode = newValue.dialCode;
          phoneWithoutCountryCode = newValue.phoneNumber;
        },
        inputDecoration: const InputDecoration(

          hintText: sPhoneHint,
          floatingLabelBehavior: FloatingLabelBehavior.always,

        ),
      ),
    );

**Comment by http600** [comment]
very quick one, after upgraded to flutter 2.10.1 from 2.8, it is not compiling for iOS, says
```
Showing Recent Messages
module map file '/Users/Name/Library/Developer/Xcode/DerivedData/Runner-bmlreonrtgngmzcywijhfadinkor/Build/Products/Release-iphoneos/libphonenumber_plugin/libphonenumber_plugin.modulemap' not found
```
I'm not sure it is just me or not, so just quick one

**Comment by frankymbieleu** [comment]
I have this problem when i had this package

warning: Multiple targets match implicit dependency for linker flags '-framework Reachability'. Consider adding an explicit dependency on the intended target to resolve this ambiguity. (in target 'Runner' from project 'Runner')
    ../../Applications/flutter/.pub-cache/hosted/pub.dartlang.org/intl_phone_number_input-0.7.1/lib/src/widgets/input_widget.dart:323:29: Error: Method 'addPostFrameCallback' cannot be called on 'WidgetsBinding?' because it is potentially null.

- 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('../../Applications/flutter/packages/flutter/lib/src/widgets/binding.dart').
    Try calling using ?. instead.
        WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
                                ^^^^^^^^^^^^^^^^^^^^

**Comment by joshpachner** [comment]
Hey love the work you're doing!

Is it possible to change the icon/values displayed in the selector?
IE instead of it being a flag + the dialCode.
Is it possible to just be the isoCode with a chevron downarrow?

Please see example in image one.
Image 2 is the best if gotten it so far.

If this is something that is not possible, but open to adding the capability then I will be down to collab and do it 

![image](https://user-images.githubusercontent.com/65133879/199831715-2920c3c3-f35a-40ac-b064-a0216b38fad9.png)
![image](https://user-images.githubusercontent.com/65133879/199831813-d22b3d95-f493-4f61-afa5-a3a09005a07b.png)

**Comment by talh11111111111** [comment]
Hi ;
The package is very nice, first of all thanks for it. I have a problem ! When I choose Germany, it only accepts a 6-digit phone number from me. I want to call a longer number. how can i solve this?

**Comment by jwag5** [comment]
Hello, great package, thank you!

I think I have found an issue but wanted to inquire first. It seems that changing the textStyle parameter does not affect the text for the dial code, just for the associated text box. So, if I have a black background, I cannot see the "+1" for example for US. Is there a way to update that Text, or is this a bug / missing feature? Thanks!

**Comment by PritSavaliya1** [comment]
I Want to Change Country Code Text Color to White  How i Change ?

**Reply by Davut-Atajanov** [reply]
selectorTextStyle: const TextStyle(color: Colors.red),

**Comment by quochuynh67** [comment]
@natintosh Hi could you help me to check the PR, I want to add more optional widget between SelectorButton and TextFormField (for example: Vertical grey line)

**Comment by muhammadfazilkp** [comment]
Swift Compiler Error (Xcode): Cannot find type 'PhoneNumberKit' in scope
/Users/muhammadfasil/.pub-cache/hosted/pub.dev/libphonenumber_plugin-0.3.3/ios/Classes/SwiftLibphonenumberPlugin.swift:6:24

Could not build the application for the simulator.
Error launching application on iPhone 16.

I am getting this error in iOS. How can I fix this?
